docs/platform-grouping/techne/developer-experience.md

```
@@ -34,9 +34,79 @@ The primary resource is `pre-commit-config` — a `.pre-commit-config.yaml` file
+
+Run the image in interactive mode.
+
+```bash
+docker run -it ghcr.io/osinfra-io/ubuntu:latest
+```
+
+To attach VS Code to the running container, install the [Dev Containers](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers) extension, click the open a remote window icon in the status bar, and select **Attach to Running Container**.
+
+This same image can be used as a devcontainer — open the repo containing a `.devcontainer/devcontainer.json` that references `ghcr.io/osinfra-io/ubuntu` in a Codespace and GitHub will use it automatically.
+
+### Codespace
```

markdownlint-cli2 (0.22.1)

[warning] Multiple headings with the same content

(MD024, no-duplicate-heading)